src/ImageSharp.Web/Processors/WebProcessingExtensions.cs

```diff
@@ -33,15 +34,27 @@ internal static class WebProcessingExtensions
             CommandParser commandParser,
             CultureInfo culture)
         {
-            if (commands.Count != 0)
+            foreach (IImageWebProcessor processor in processors.GetBySupportedCommands(commands.Keys.ToList()))
             {
-                foreach (IImageWebProcessor processor in processors)
-                {
-                    source = processor.Process(source, logger, commands, commandParser, culture);
-                }
+                source = processor.Process(source, logger, commands, commandParser, culture);
             }
 
             return source;
         }
+
+        /// <summary>
+        /// Sorts the processors according to the first supported command and removes processors without supported commands.
+        /// </summary>
+        /// <param name="processors">The collection of available processors.</param>
+        /// <param name="commands">The list of parsed commands.</param>
+        /// <returns>
+        /// The sorted proccessors.
+        /// </returns>
+        public static IEnumerable<IImageWebProcessor> GetBySupportedCommands(this IEnumerable<IImageWebProcessor> processors, IList<string> commands)
+            => processors
+                .GroupBy(p => p.Commands.Min(c => (int?)commands.IndexOf(c)) ?? -1) // Get index of first supported command
```

Comment:
We don't expect/support `Commands` to return `null`, but it could return an empty collection: in that case we fallback to -1 and filter out the processor. This means it's not possible to create a processor that is called for all requests.

We could default to 0 (that would run it as one of the first processors though) or introduce a new property on the `IImageWebProcessor` to provide this value, any thoughts?

Reply:
Lets not complicate things any further (and introduce a breaking change by adding a new property to the `IImageWebProcessor`): if you want a processor to be called on every request, you can add the command in the `OnParseCommandsAsync` callback, in a custom `IRequestParser` or even manually do any processing in the `OnProcessedAsync` callback.